keyboards/phantagom/oled_mpad/keymaps/via/keymap.c

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright 2022 <dennis@kruyt.org>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+enum Layer {
+    _NUMPAD = 0,        // Standard QWERTY layer
+    _MOUSE,          // Function key layer
+    _NAV,
+};
+#define _NP _NUMPAD
+#define _MS _MOUSE
+#define _NV _NAV
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  /*
+   * ,-------------------.
+   * | Fx | Fx | Fx | Fx |
+   * |----|----|----|----|
+   * | NL | /  | *  | -  |
+   * |----|----|----|----|
+   * | 7  | 8  | 9  |    |
+   * |----|----|----| +  |
+   * | 4  | 5  | 6  |    |
+   * |----|----|----|----|
+   * | 1  | 2  | 3  |    |
+   * |----|----|----| En |
+   * |    0    | .  |    |
+   * `-------------------'
+   */
+
+  [_NUMPAD] = LAYOUT_numpad_6x5(
+    MO(_MS), KC_MUTE, KC_VOLD, KC_VOLU,
+    KC_PSCR, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS, 
+    TO(_NP), KC_P7,   KC_P8,   KC_P9,   
+    TO(_MS), KC_P4,   KC_P5,   KC_P6,   KC_PPLS, 
+    TO(_NV), KC_P1,   KC_P2,   KC_P3,   
+    KC_A, KC_P0, KC_PDOT,  KC_PENT  
+  ),
+
+  [_MOUSE] = LAYOUT_numpad_6x5(
+    _______, KC_MUTE, KC_VOLD, KC_VOLU,
+    KC_PSCR, KC_NUM, KC_BTN1, KC_BTN2, KC_BTN3, 
+    TO(_NP), KC_P7,   KC_MS_U,   KC_P9,   
+    TO(_MS), KC_MS_L,   KC_P5,   KC_MS_R,   KC_PPLS, 
+    TO(_NV), KC_P1,   KC_MS_D,   KC_P3,   
+    KC_A, KC_P0, KC_PDOT,  KC_PENT
+  ),
+
+  [_NAV] = LAYOUT_numpad_6x5(
+    MO(_MS), KC_MUTE, KC_VOLD, KC_VOLU,
+    KC_PSCR, KC_NUM, KC_PSLS, KC_PAST, KC_PMNS, 
+    TO(_NP), KC_HOME,   KC_UP,   KC_PGUP,   
+    TO(_MS), KC_LEFT,   _______,   KC_RIGHT,   KC_PPLS, 
+    TO(_NV), KC_END,   KC_DOWN,   KC_PGDOWN,   
+    KC_A, KC_INS, KC_DEL,  KC_PENT  
+  ),
+};
+
+void matrix_init_user(void) {
+  // set CapsLock LED to output and low
+  setPinOutput(GP4);
+  writePinLow(GP4);
+  // set NumLock LED to output and low
+  setPinOutput(GP5);
+  writePinLow(GP5);
+  // set ScrollLock LED to output and low
+  setPinOutput(GP6);
+  writePinLow(GP6);
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        writePin(GP4, led_state.num_lock);
+        writePin(GP5, led_state.caps_lock);
+        //writePin(B2, led_state.scroll_lock);
+    }
+    return res;
+}
```

Comment:
Same as https://github.com/qmk/qmk_firmware/pull/18493/files#r990802288 :
```suggestion
void led_update_ports(led_t led_state) {
    writePin(GP4, led_state.num_lock);
    writePin(GP5, led_state.caps_lock);
    //writePin(B2, led_state.scroll_lock);
}
```
Note that this implementation nearly matches the default implementation. Why not set the right pin definitions (`LED_NUM_LOCK_PIN`, `LED_CAPS_LOCK_PIN`, but apparently not `LED_SCROLL_LOCK_PIN` or any others)?

Reply:
changed to led_update_port and enabled scroll lock, dunno why I did not enabled it before...